### PR TITLE
BLOCKER - [WFCORE-3859] Add transformer to transform from version  2.0.0 to 1.2.0 of the Elytron subsystem model.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.jboss.org/browse/WFCORE-3859

### CODE REVIEW

**Comment by jmesnil** [comment]
Is there a transformer test covering this change?
I didn't see a elytron-transformers-2.0.xml or elytron-transformers-2.0-reject.xml in the test resources

**Comment by darranl** [comment]
Just looking to clean something up in the testsuite - we have an elytron-transformers-1.2.xml and the reject pair as that is what we are testing we are transforming to - but there seem to some mixed code in the test case that doesn't actually test anything that I am looking into.

**Comment by darranl** [comment]
Retest this please

File: elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2-reject.xml [hunk 1]
```diff
@@ -1,45 +1,5 @@
 <subsystem xmlns="urn:wildfly:elytron:3.0">
```
Comment (jmesnil) [inline elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2-reject.xml]: shouldn't the XML namespace be `urn:wildfly:elytron:1.2`?

File: elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2.xml [hunk 2]
```diff
@@ -16,6 +16,9 @@
     <security-realms>
         <identity-realm name="identity" identity="identity" />
     </security-realms>
+    <credential-security-factories>
```
Comment (jmesnil) [inline elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2.xml]: same remark as above on the xml namespace

File: elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2-reject.xml (hunk 1 above)
Comment (darranl) [reply elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2-reject.xml]: No as I already said above - the version in this name is the version we are transforming to.

File: elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2.xml (hunk 2 above)
Comment (darranl) [reply elytron/src/test/resources/org/wildfly/extension/elytron/elytron-transformers-1.2.xml]: No as I already said above - the version in this name is the version we are transforming to.

**Comment by darranl** [comment]
The naming convention was set in this PR https://github.com/wildfly/wildfly-core/pull/3002 and approved by @bstansberry - at the time the elytron-transformers-1.2.xml was using namespace 2.0.

I would prefer we test transformations to the versions of WildFly Core we release but a decision was made to test only to the versions used in EAP.

**Comment by bstansberry** [comment]
I don't care much about file names, but transformers are testing transformation to API versions not xsd versions, so if a file is used for such a test naming it after the target API version makes sense to me.

Re; "test only to the versions used in EAP", I think that's about API versions 1.0 and 1.1?  Those were in Core 3.0.0 and 3.0.2 respectively, with 3.0.6 introducing version 1.2.  Core 3.0.8 was what was used in WildFly 11.0.0.Final.  So not testing 1.0 and 1.1 doesn't mean we only care about EAP, it just means we are ignoring core versions that were never used in a WildFly Final.

In a perfect world we'd test transforming to 1.0 and 1.1 too. I wrote the tests that #3002 removed! And they passed!  But IIRC there was some hassle related to them or something, or maybe I just got tired of fighting over it.

**Pull request merged**